### ISSUE TITLE
Setup registry for Engine Image

### ISSUE THREAD

**Issue by sipsma** [dagger/dagger#3650]
Quoting @aluzzardi : 

Chatted with @samalba about ECR and so forth.

So far my preference would be:

1) Vanity URL + ECR.
- With a vanity URL users don't see the "real" registry address, and that's the only thing Docker Hub has going on
- Vanity requires infrastructure which will be on AWS, might as well have the registry there and not pay traffic between the 2
- ECR is cheaper and way more stable than Hub

2) IF vanity URLs don't work ... I'd suggest we go with GitHub
- We're already "leaking" github.com/dagger/dagger, might as well also leak ghcr.io/dagger if vanity doesn't work
- I trust the stability more

3) Hub: this would be last preference, depending on the startup program
- If vanity doesn't work, we get a "vanity-ish" image (hub short name)

---

Related open PR for creating that image: https://github.com/dagger/dagger/pull/3647

**Comment by gerhard** [comment]
How does this issue relate to #3347?

FWIW, I must be missing context - was away last week. Where should I look?

**Comment by sipsma** [comment]
@gerhard There's some more context here: https://github.com/dagger/dagger/issues/3653

tl;dr is that the engine image won't be used to distribute our CLI or be related at all. It *will* be used to distribute a helper binary the SDKs need for now, but that binary is separate from the CLI and 100% internal, not for human consumption.

**Comment by gerhard** [comment]
If we were to use a vanity URL for our registry, which of the following would you prefer? cc @shykes @samalba @aluzzardi @sipsma

- dg.cr/engine (need to buy)
- dagger.cr/engine (need to buy)
- dgcr.io/engine (need to buy)
- r.dagger.io/engine
- registry.dagger.io/engine
- images.dagger.io/engine

**Comment by sipsma** [comment]
> registry.dagger.io/engine

Is my vote. It's the most obvious and as of right now we don't need to optimize for easy type-ability (these images are only ever pulled automatically).

If we do want to optimize for type-ability right away (for the hypothetical future where users need to type it a lot), then `r.dagger.io` works. It's easier to remember and more recognizable relative to `dg.cr`/`dgcr.io`/etc. IMO.

**Comment by shykes** [comment]
I'm OK with any of these names. Slight preference for `registry.dagger.io`. My only question is: is there any scenario where we would need to host a non-OCI registry? Ie language packages. If the answer is yes, maybe we need something more oci-specific, to leave that door open? Like `oci.dagger.io` or `containers.dagger.io` ?
